### PR TITLE
MDP-206: Homepage dynamic rendering of articles

### PR DESCRIPTION
**Pull request by Stephaaan**

### CODE REVIEW

File: src/components/Layout/index.tsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
```
Comment (garySiladi) [inline src/components/Layout/index.tsx]: I'm not sure if you're still working on this file. The children don't need to be typed, just use the FC type of React.
```
import React, { FC } from "react";
import { Header } from "../Header/Header";
import { Footer } from "../Footer";

export const Layout: FC = ({ children }) => (
  <>
    <Header />
    {children}
    <Footer />
  </>
);
```

File: src/components/Header/Header.tsx [hunk 2]
```diff
@@ -14,18 +14,22 @@ export const Header = () => {
   return (
     <div>
       {openMenu && (
-        <div>
-          <div className="flex justify-end mr-24 mt-16">
-            <div onClick={() => setOpenMenu(false)}>
-              <ImageHeader src={closeImg} />
+        <>
```
Comment (OlehVynnyk) [inline src/components/Header/Header.tsx:17]: When clicking at the user icon, user profile work incorrect.

File: src/components/Header/Header.tsx [hunk 3]
```diff
@@ -14,18 +14,22 @@ export const Header = () => {
...
+          <div className="lg:hidden bg-white z-50 absolute w-full">
+            <div className="flex justify-end mr-24 mt-16">
+              <div onClick={() => setOpenMenu(false)}>
+                <ImageHeader src={closeImg} />
+              </div>
+            </div>
+            <div className="z-50">
+              <HamburgerMenu />
             </div>
           </div>
```
Comment (OlehVynnyk) [inline src/components/Header/Header.tsx:27]: "Arteterapie", "Rozvrh" and "Clanky" has not correctly position

File: src/components/Homepage/Button.tsx [hunk 4]
```diff
@@ -7,11 +7,12 @@ interface IProps {
...
 export const Button: FC<IProps> = ({
   children,
-  className = "text-white font-lato font-bold"
+  className = "text-white font-lato font-bold "
 }) => (
   <button
+    style={{ lineHeight: "1" }}
     className={classNames(
-      "w-full h-40 flex justify-center items-center",
+      "w-full h-40 flex justify-center items-center items-center",
```
Comment (OlehVynnyk) [inline src/components/Homepage/Button.tsx]: Change button font

File: src/components/Header/Header.tsx [hunk 5]
```diff
@@ -14,42 +14,50 @@ export const Header = () => {
   return (
     <div>
       {openMenu && (
-        <div>
-          <div className="flex justify-end mr-24 mt-16">
-            <div onClick={() => setOpenMenu(false)}>
-              <ImageHeader src={closeImg} />
+        <>
```
Comment (OlehVynnyk) [inline src/components/Header/Header.tsx:17]: Rename Header.tsx to index.tsx

File: src/components/Homepage/Button.tsx [hunk 6]
```diff
@@ -7,9 +7,10 @@ interface IProps {
 
 export const Button: FC<IProps> = ({
   children,
-  className = "text-white font-lato font-bold"
+  className = "text-white font-lato font-bold "
```
Comment (OlehVynnyk) [inline src/components/Homepage/Button.tsx]: delete space at the end of div

File: src/components/Homepage/LandingPageWrapper.tsx [hunk 7]
```diff
@@ -25,10 +25,11 @@ export const LandingPageWrapper = () => {
         <Description>
           Vyskúšajte si CHAT. Registrujte sa a získajte prvú hodinu zdarma.
         </Description>
-        <div className="w-192 m-auto mt-24 bg-black rounded-8">
+        <div className="w-192 m-auto mt-24 bg-black rounded-8 ">
```
Comment (OlehVynnyk) [inline src/components/Homepage/LandingPageWrapper.tsx]: same problem with space

**Review by OlehVynnyk — DISMISSED** [review]

File: src/components/Homepage/NewestArticles.tsx [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  heading: string;
+  insight: string;
+  title_photo: string;
+}
+
+export const NewestArticles = () => {
+  const [articles, setArticles] = useState([]);
+  const [loadingError, setLoadingError] = useState(false);
+  useEffect(() => {
+    fetch("http://localhost/api/article/newest")
```
Comment (garySiladi) [inline src/components/Homepage/NewestArticles.tsx]: don't we have a real endpoint running?

Comment (garySiladi) [reply src/components/Homepage/NewestArticles.tsx]: also have the given endpoint (`http://localhost/api/`) from a config file imported

File: src/components/Homepage/NewestArticles.tsx [hunk 9]
```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useState } from "react";
+import icon from "../../static/svg/homer.svg";
+import { Thumbnail } from "../Thumbnail/Thumbnail";
+import { Button } from "../Button";
+
+interface IArticle {
+  id: number;
+  heading: string;
+  insight: string;
+  title_photo: string;
```
Comment (garySiladi) [inline src/components/Homepage/NewestArticles.tsx:11]: naming should be `photoTitle`

File: src/components/Homepage/NewestArticles.tsx [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+      <div className="flex flex-col items-center justify-center lg:flex-row">
+        {articles.length === 0 ? (
+          loadingError ? (
+            <div>Server error</div>
+          ) : (
+            <div>Loading articles...</div>
+          )
+        ) : (
+          articles.map((article: IArticle) => (
+            <div
```
Comment (garySiladi) [inline src/components/Homepage/NewestArticles.tsx]: create an article component in a separate file which instances you would return from this .map

Comment (garySiladi) [reply src/components/Homepage/NewestArticles.tsx]: that new file should contain the IArticle interface

File: src/components/Homepage/NewestArticles.tsx [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+      .then(response => response.json())
+      .then(response => setArticles(response))
+      .catch(error => setLoadingError(true));
+  }, []);
+  return (
+    <div className="m-0 bg-white relative pt-48 lg:pt-24 lg:m-48 text-center pl-32 pr-32 lg:pl-128 lg:pr-128 shadow">
+      <h1 className="font-gilbert text-31 mb-16">Najnovšie články</h1>
+      <div className="flex flex-col items-center justify-center lg:flex-row">
+        {articles.length === 0 ? (
+          loadingError ? (
```
Comment (garySiladi) [inline src/components/Homepage/NewestArticles.tsx]: don't have nested conditions + defensive programming, so first you solve the exception cases

**Review by garySiladi — CHANGES_REQUESTED** [review]

File: src/components/Homepage/Button.tsx [hunk 12]
```diff
@@ -10,6 +10,7 @@ export const Button: FC<IProps> = ({
   className = "text-white font-lato font-bold"
 }) => (
   <button
+    style={{ lineHeight: "1" }}
```
Comment (pggl) [inline src/components/Homepage/Button.tsx]: use tailwind

File: src/components/Homepage/NewestArticles.tsx (hunk 8 above)
Comment (pggl) [reply src/components/Homepage/NewestArticles.tsx]: we need create env file

File: src/components/Homepage/NewestArticles.tsx [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+              style={{ height: "408px" }}
+              key={article.id}
+              className="w-238 ml-0 mt-8 lg:mt-0 lg:ml-8"
+            >
+              <Thumbnail
+                name={article.heading}
+                icon={
+                  "http://localhost/api/image/214/160/" + article.title_photo
+                }
+                alt="super"
```
Comment (pggl) [inline src/components/Homepage/NewestArticles.tsx]: why is "super" hardcoded?

**Review by pggl — DISMISSED** [review]

File: src/components/Homepage/NewestArticles.tsx (hunk 9 above)
Comment (Stephaaan) [reply src/components/Homepage/NewestArticles.tsx:11]: this is name that's returned from backend (this is default laravel convention), so, should I leave it like that or add a few lines to code to rename that key?

File: src/components/Homepage/NewestArticles.tsx (hunk 13 above)
Comment (Stephaaan) [reply src/components/Homepage/NewestArticles.tsx]: because we don't have a field for title photo description on backend. Is it necessary?

Comment (Stephaaan) [reply src/components/Homepage/NewestArticles.tsx]: Is it okay to add as alt sentence: The title photo of article ${article.title}?

File: src/components/Homepage/NewestArticles.tsx (hunk 9 above)
Comment (pggl) [reply src/components/Homepage/NewestArticles.tsx:11]: rename it

File: src/components/Homepage/NewestArticles.tsx (hunk 13 above)
Comment (pggl) [reply src/components/Homepage/NewestArticles.tsx]: maybe article name is good for alt

**Review by garySiladi — APPROVED** [review]

**Pull request merged**